### PR TITLE
UML 951 - want to understand when an LPA can be used

### PR DESCRIPTION
**Pull request by SeemaMenon**
## Purpose
https://opgtransform.atlassian.net/browse/UML-951

Fixes UML-951

- Summary handler amended to pass actor details parameter to make content display decision in twig based on actor type
- Fixed broken tests

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Refunds service_

## Checklist

* [X] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-front/app/features/context/Integration/AccountContext.php [hunk 1]
```diff
@@ -673,7 +673,7 @@ public function iRequestToViewAnLPAWhichStatusIs($status)
      */
     public function theFullLPAIsDisplayedWithTheCorrect($message)
     {
-        $lpa = $this->lpaService->getLpaById($this->userIdentity, $this->actorLpaToken);
+        list($lpa, $actorLpa) = $this->lpaService->getLpaById($this->userIdentity, $this->actorLpaToken);
```
Comment (cooperaj) [inline service-front/app/features/context/Integration/AccountContext.php:676]: Would like a quick chat about these changes. They read ok and I can see why they were needed but I'm not sure how our tests were working without them :S

File: service-front/app/features/context/UI/AccountContext.php [hunk 2]
```diff
@@ -1128,8 +1128,8 @@ public function iRequestToGiveAnOrganisationAccessToOneOfMyLPAs()
                     json_encode([
                         'user-lpa-actor-token' => $this->userLpaActorToken,
                         'date'                 => 'date',
-                        'lpa'                  => $this->lpa,
-                        'actor'                => $this->lpaData['actor'],
+                        'lpa'  => $this->lpa,
```
Comment (cooperaj) [inline service-front/app/features/context/UI/AccountContext.php:1131]: Please retain this kind of indentation as it improves the readability of the code

File: service-front/app/src/Common/src/Service/Lpa/LpaService.php [hunk 3]
```diff
@@ -78,10 +78,10 @@ public function getLpas(string $userToken): ?ArrayObject
     /**
      * @param string $userToken
      * @param string $actorLpaToken
-     * @return Lpa|null
+     * @return array
      * @throws Exception
      */
-    public function getLpaById(string $userToken, string $actorLpaToken): ?Lpa
+    public function getLpaById(string $userToken, string $actorLpaToken): ?array
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Service/Lpa/LpaService.php:84]: I think we need to have a refactor of the way we do DTO's. Bunging everything in an array is not the most ideal solution.

File: service-front/app/features/context/UI/AccountContext.php (hunk 2 above)
Comment (SeemaMenon) [reply service-front/app/features/context/UI/AccountContext.php:1131]: My bad....correcting every one on the page now

File: service-front/app/src/Common/src/Service/Lpa/LpaService.php (hunk 3 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Service/Lpa/LpaService.php:84]: @cooperaj  - Can we discuss on this?

Comment (SeemaMenon) [reply service-front/app/src/Common/src/Service/Lpa/LpaService.php]: Changed to return an array object for now as agreed. A new ticket created for building dto's for the same

File: service-front/app/src/Actor/templates/actor/view-lpa-summary.html.twig [hunk 4]
```diff
@@ -45,11 +45,20 @@
 
             <div class="govuk-grid-row lpa-valid-border">
                 <div class="govuk-grid-column-full column-full-alt-1">
-                    <h2 class="govuk-heading-m heading-m-alt-1">This LPA is {{ lpa.status|lower }}</h2>
+                    {% if lpa.caseSubtype == "pfa" and lpa.attorneyActDecisions == 'When Registered' %}
```
Comment (GemTay) [inline service-front/app/src/Actor/templates/actor/view-lpa-summary.html.twig:48]: Change lpa data to lowercase for comparison so it is not case sensitive `lpa.attorneyActDecisions|lower == 'when registered'`

File: service-front/app/src/Actor/templates/actor/view-lpa-summary.html.twig [hunk 5]
```diff
@@ -174,9 +183,14 @@
                     {% if lpa.caseSubtype == "pfa" %}
                         <div class="govuk-summary-list__row">
                             <dt class="govuk-summary-list__key">When can the LPA be used?</dt>
+                            {{ actor.actorType }}
```
Comment (GemTay) [inline service-front/app/src/Actor/templates/actor/view-lpa-summary.html.twig:186]: I don't think the actor type needs to be displayed here

Comment (SeemaMenon) [reply service-front/app/src/Actor/templates/actor/view-lpa-summary.html.twig:186]: MY BADDDDDD..... was part of testing. Good catch Gemma. :)

File: service-front/app/src/Actor/templates/actor/view-lpa-summary.html.twig (hunk 4 above)
Comment (SeemaMenon) [reply service-front/app/src/Actor/templates/actor/view-lpa-summary.html.twig:48]: done..TA

File: service-front/app/src/Actor/templates/actor/view-lpa-summary.html.twig [hunk 6]
```diff
@@ -175,9 +184,13 @@
                         <div class="govuk-summary-list__row">
                             <dt class="govuk-summary-list__key">When can the LPA be used?</dt>
                             <dd class="govuk-summary-list__value">
-                                {% if lpa.attorneyActDecisions == 'When Registered' %}
-                                    While the donor has mental capacity, this <abbr title="lasting power of attorney">LPA</abbr> should only be used with the donor’s permission
-                                {% elseif lpa.attorneyActDecisions == 'Loss of capacity' %}
+                                {% if lpa.attorneyActDecisions|lower == 'when registered' %}
+                                    {% if actor.type == 'donor' %}
+                                        You have said that this LPA can be used now. While you have mental capacity, your attorneys can only use the LPA if you allow them to.
```
Comment (cooperaj) [inline service-front/app/src/Actor/templates/actor/view-lpa-summary.html.twig:189]: Please add `<abbr title="lasting power of attorney">LPA</abbr>` in place of 'LPA' in all cases

File: service-front/app/src/Common/src/Service/Lpa/LpaService.php [hunk 7]
```diff
@@ -78,28 +78,28 @@ public function getLpas(string $userToken): ?ArrayObject
...
      */
-    public function getLpaById(string $userToken, string $actorLpaToken): ?Lpa
+    public function getLpaById(string $userToken, string $actorLpaToken): ?ArrayObject
     {
         $this->apiClient->setUserTokenHeader($userToken);
 
         $lpaData = $this->apiClient->httpGet('/v1/lpas/' . $actorLpaToken);
 
-        $lpa = isset($lpaData['lpa']) ? $this->lpaFactory->createLpaFromData($lpaData['lpa']) : null;
+        if (isset($lpaData['lpa'])) {
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Service/Lpa/LpaService.php:90]: Please retain the original code structure here. What has been introduced is an increase in complexity.

File: service-front/app/src/Common/src/Service/Lpa/LpaService.php [hunk 8]
```diff
@@ -78,28 +78,28 @@ public function getLpas(string $userToken): ?ArrayObject
...
-                    'uId' => $lpa->getUId()
+                    'id' => $userToken,
+                    'uId' => $lpaData['lpa']->getUId()
                 ]
             );
+            return $lpaData;
         }
-
-        return $lpa;
+        return null;
```
Comment (GemTay) [inline service-front/app/src/Common/src/Service/Lpa/LpaService.php:102]: If null is returned the handler needs added functionality to handle that case

File: service-front/app/src/Common/src/Service/Lpa/LpaService.php (hunk 7 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Service/Lpa/LpaService.php]: Done

File: service-front/app/src/Common/src/Service/Lpa/LpaService.php (hunk 8 above)
Comment (SeemaMenon) [reply service-front/app/src/Common/src/Service/Lpa/LpaService.php]: Restructured ^

**Pull request merged**